packages/components/src/internal/components/navigation/FolderMenu.tsx

```diff
@@ -0,0 +1,98 @@
+import React, { FC, memo, useCallback, useEffect, useState } from 'react';
+import { ActionURL } from '@labkey/api';
+import { Dropdown, MenuItem } from 'react-bootstrap';
+
+import {
+    Alert,
+    AppContext,
+    buildURL,
+    isLoading,
+    LoadingSpinner,
+    LoadingState,
+    resolveErrorMessage,
+    useAppContext,
+    useServerContext,
+} from '../../..';
+
+import { getCurrentAppProperties } from '../../app/utils';
+import { blurActiveElement } from '../../util/utils';
+import { AppProperties } from '../../app/models';
+
+interface FolderMenuItem {
+    href: string;
+    id: string;
+    label: string;
+}
+
+interface Props {
+    appProperties?: AppProperties;
+}
+
+export const FolderMenu: FC<Props> = memo(({ appProperties }) => {
+    const { controllerName } = appProperties ?? getCurrentAppProperties();
+    const [error, setError] = useState<string>();
+    const [loading, setLoading] = useState<LoadingState>(LoadingState.INITIALIZED);
+    const [open, setOpen] = useState(false);
+    const [items, setItems] = useState<FolderMenuItem[]>([]);
+    const hasError = !!error;
+    const isLoaded = !isLoading(loading);
+    const { api } = useAppContext<AppContext>();
+    const { container } = useServerContext();
+
+    useEffect(() => {
+        setLoading(LoadingState.LOADING);
+        setError(undefined);
+
+        (async () => {
+            try {
+                const folders = await api.security.fetchContainers({
+                    containerPath: container.type === 'project' ? container.path : container.parentPath,
+                });
+
+                const items_: FolderMenuItem[] = folders.map(folder => ({
```

Comment:
Q: what does the nomenclature of the trailing underscore mean?

Reply:
This is a way of denoting a scoped variable without overriding the name of variable in the external scope. `items` is a variable declared outside of the anonymous function which is brought in via closure. Here, I name the variable internal the method, which represents the same thing as `items`, with an underscore appended. If I used the same name it results in a warning which is a code smell for potentially erroneous logic. Here is an example if I had named the internal variable also `items`:

![image](https://user-images.githubusercontent.com/3926239/146815080-aa7dfed4-839c-4d6e-a04e-5d008c8d548b.png)